packages/context/src/objects/objects.ts

```diff
@@ -72,38 +134,164 @@ export type SelectOption = {
     value: string;
 };
 
+export type VisibilityConfiguration = {
+    operator?: 'any' | 'all';
+    conditions?: {
+        property: string;
+        operator: 'eq' | 'neq';
+        value: string | number;
+    }[];
+};
+
+export type DisplayConfiguration = {
+    label?: string;
+    placeholder?: string;
+    required?: boolean;
+    description?: string;
+    defaultValue?: string | number | string[];
+    readOnly?: boolean;
+    tooltip?: string;
+    prefix?: string;
+    suffix?: string;
+    placeholderChar?: string;
+    rowCount?: number;
+    charCount?: boolean;
+    mode?: 'default' | 'existingOnly';
+    relatedObjectDisplay?: 'dropdown' | 'dialogBox';
+    visibility?: VisibilityConfiguration | string;
+};
+
+export type InputParameterReference = {
+    type: 'input';
+    parameterId: string;
+    display?: DisplayConfiguration;
+    enumWithLabels?: SelectOption[];
+};
+
+export type Content = {
+    type: 'content';
+    html: string;
+    visibility?: VisibilityConfiguration | string;
+};
+
+export type Column = {
+    width: number;
+    entries?: FormEntry[];
+};
+
+export type Columns = {
+    type: 'columns';
+    label?: string;
+    columns: Column[];
+    visibility?: VisibilityConfiguration | string;
+};
+
+export type Section = {
+    label: string;
+    entries?: FormEntry[];
+};
+
+export type Sections = {
+    type: 'sections';
+    label?: string;
+    sections: Section[];
+    visibility?: VisibilityConfiguration | string;
```

Comment:
I see this is how it is in the services, but why is there `| string` here?

Reply:
FormIO supports more complex conditions in JSON format - which is saved as a string.

Reply:
If it is a JSON string, why wouldn't it be saved as an object instead?  Does FormIO actually expect a JSON string that it then parses into some object?

Reply:
Yeah, they expect a string. I'm guessing they parse it behind the scenes.

We could technically save it as an object in our database, and "stringify" it when converting to the FormIO components.

Reply:
It's fine.  In any case I wasn't expecting this to change that anyway since it requires services to change, was just curious why it was a string.  But if that's what FormIO expects, I think we can just leave it.